frontend/src/app/services/simulation.service.ts

```diff
@@ -67,13 +70,18 @@ export class SimulationService {
               break;
           }
 
+          this.edgeService.emitEdges();
           this.visualisationService.emitGraph(this.graph);
           this.minerService.emit();
         }
         this.stepService.unblockSemaphore();
       })
     ).subscribe();
 
+    this.edgeService.getEdges().subscribe(() => {
+      this.visualisationService.emitGraph(this.graph);
+    })
+
```

Comment:
eeeeczy czy to nie jest troche bez sensu? bo tak jakby 2 razy graf emitujemy w 1 iteracji

Reply:
chyba ze zle to rozumiem

Reply:
nie no właśnie ja to tak rozumiem, że tam wyżej emitujemy graf w razie nowego stepa, a w tym niższym subskrybujemy się na nowo dodane edge i jak dostaniemy coś nowego to update'ujemy graf

Reply:
tak robimy^

Reply:
troszkę to zmieniam, żeby częściej się to aktualizowało